docs/Design-and-Modeling/10-api-security.md

```diff
@@ -0,0 +1,111 @@
+
+### API Keys
+
+For API keys, select the _apiKey_ scheme type. An API key is a single token sent to the API server that the user has retrieved through an out-of-band mechanism. API keys can be a query parameter, a custom HTTP header, or a cookie.
+
+![API Key Configuration](../../assets/images/security-apikey.png)
+
+Use the right dropdown to toggle between _query_, _header_, and _cookie_ and use the name textbox to enter the query, cookie, or header name, respectively.
+
+When designing a new API with API Keys, we recommend the following:
+
+- Use _cookie_ when a web application uses your API in the browser, and you can use cookies for session management.
+- Use _query_ when your API is mostly exposing public data or doesn't have strict security constraints and may be used in environments where developers can only specify a URL but no custom headers.
+- In all other cases, use _header_ as the best practice that follows proper HTTP semantics.
+
+### HTTP Authentication
+
+For HTTP authentication, select the _http_ scheme type. HTTP authentication uses the standard HTTP _Authorization_ header defined in [RFC7235](https://tools.ietf.org/html/rfc7235).
+
+![HTTP Auth Configuration](../../assets/images/security-http.png)
+
+Use the right dropdown to switch between the different HTTP authentication schemes. OpenAPI and, by extension, Stoplight support all the [schemes registered at IANA](https://www.iana.org/assignments/http-authschemes/http-authschemes.xhtml). Further configuration is neither necessary nor possible.
+
+While OpenAPI supports the different variants of HTTP authentication, the majority of APIs use _basic_. With basic authentication, an API client sends the combination of a username and a password in the Authorization header. The method for retrieving the username and password is not part of the OpenAPI description.
+
+### OAuth 2
+
+For OAuth 2, select the _oauth2_ scheme type. OAuth is an open authentication standard defined by [RFC6749](https://tools.ietf.org/html/rfc6749) and [RFC6750](https://tools.ietf.org/html/rfc6750). OAuth also uses the standard HTTP _Authorization_ header to send a credential known as bearer token but also defines various flows for retrieving this bearer token. You can enter the configuration for the flows into your OpenAPI specification, and various OpenAPI tools can use this information to assist the user in going through them.
+
+Hint: If you are documenting an API that still uses OAuth 1, select the _http_ scheme type and then _oauth_ as a subtype instead.
+
+![OAuth Configuration](../../assets/images/security-oauth-a.png)
+
+Once selecting _oauth2_, you will see a flows input field into which you can enter one or more of the following flow types: _implicit_, _password_, _clientCredentials_, and _authorizationCode_. An autocomplete dropdown appears as soon as you click in the flows field.
```

Comment:
Maybe we can build upon this by talking a little about when to use different flows? Taking things from here: https://auth0.com/docs/api-auth/which-oauth-flow-to-use

Also maybe we can move OAuth 2 before HTTP?

Reply:
@philsturgeon What do you think?

Reply:
Changed the order and included a very brief description of the flows.